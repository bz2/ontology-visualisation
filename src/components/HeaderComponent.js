import { useRef } from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { ProgressSpinner } from 'primereact/progressspinner'
import { Button } from 'primereact/button'
import { OverlayPanel } from 'primereact/overlaypanel'
import { useRouter } from 'next/router'
import { useTranslation } from 'react-i18next'
import logo from '../assets/images/logo.svg'
import logout from '../utils/auth/logout'
import actions from '../store/actions'
import { ROUTE_LOGIN } from '../constants/routes'

const HeaderComponent = ({
  loading,
  setStoreState,
  user
}) => {
  const overlay = useRef(null)
  const router = useRouter()
  const { t } = useTranslation()

  return (
    <header>
      <div>
        {
          loading && (
            <div className="loader-box">
              <ProgressSpinner
                className="spinner"
                strokeWidth="4"
              />
            </div>
          )
        }
        <div className="logo">
          <img
            src={logo}
            alt="Highways England"
          />
        </div>
      </div>

      <div>
        <Button
          type="button"
          icon="pi pi-user"
          onClick={(e) => overlay.current.toggle(e)}
        />

        <OverlayPanel ref={overlay}>
          {
            user.isGuest ? (
              <Button
                icon="pi pi-sign-in"
                label={t('signIn')}
                className="p-button-secondary"
                onClick={() => router.push(ROUTE_LOGIN)}
              />
            ) : (
              <Button
                icon="pi pi-sign-out"
                label={t('signOut')}
                className="p-button-secondary"
                onClick={() => logout({
                  router,
                  setStoreState
                })}
              />
            )
          }

        </OverlayPanel>
      </div>
    </header>
  )
}

HeaderComponent.propTypes = {
  loading: PropTypes.bool.isRequired,
  setStoreState: PropTypes.func.isRequired,
  user: PropTypes.shape.isRequired,
}

const mapToProps = ({
  loading,
  user
}) => ({
  loading,
  user
})

export default connect(
  mapToProps,
  actions
)(HeaderComponent)
