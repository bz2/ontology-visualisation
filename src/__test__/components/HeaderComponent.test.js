import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import HeaderComponent from '../../components/HeaderComponent'

const setup = ({
  user,
  loading
}) => {
  const props = {
    loading,
    updateStoreValue: jest.fn(),
    user
  }

  const component = shallow(<HeaderComponent {...props} />)

  return {
    component,
    props
  }
}

describe('HeaderComponent', () => {
  it('should match snapshot when user not logged in and not loading', () => {
    const {
      component
    } = setup({
      user: {
        email: '',
        isGuest: false
      },
      loading: 0
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when user logged in and loading', () => {
    const {
      component
    } = setup({
      user: {
        email: 'test@test.com',
        isGuest: false
      },
      loading: 1
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
