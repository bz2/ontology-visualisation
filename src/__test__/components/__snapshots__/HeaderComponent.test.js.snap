// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match snapshot when user logged in and loading 1`] = `
<Connect
  actions={
    Object {
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  loading={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  user={
    Object {
      "email": "test@test.com",
      "isGuest": false,
    }
  }
>
  <Component />
</Connect>
`;

exports[`HeaderComponent should match snapshot when user not logged in and not loading 1`] = `
<Connect
  actions={
    Object {
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  loading={false}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  user={
    Object {
      "email": "",
      "isGuest": false,
    }
  }
>
  <Component />
</Connect>
`;
